doc/metrics.md

```diff
@@ -0,0 +1,62 @@
+# Metrics
+
+etcd-backup-restore uses [Prometheus][prometheus] for metrics reporting. The metrics can be used for real-time monitoring and debugging. It won't persist its metrics; if a member restarts, the metrics will be reset.
+
+The simplest way to see the available metrics is to cURL the metrics endpoint `/metrics`. The format is described [here](http://prometheus.io/docs/instrumenting/exposition_formats/).
+
+Follow the [Prometheus getting started doc][prometheus-getting-started] to spin up a Prometheus server to collect etcd metrics.
+
+The naming of metrics follows the suggested [Prometheus best practices][prometheus-naming]. All etcd-backup-restore related metrics are put under namespace `etcdbr`.
+
+### ETCD metrics
+
+The metrics under the `etcd` prefix/namespace are carried forward from etcd library that we use. These metrics does not include details of the `etcd` deployment on which `etcd-backup-restore` utility works. Instead, it helps the monitoring the `embedded etcd` we spawn while restoration process.
+
+
+### Snapshot
+
+These metrics describe the status of the snapshotter. In order to detect outages or problems for troubleshooting, these metrics should be closely monitored. The below mentioned metrics are listed as collection of series using prometheus labels `kind` and `succeeded`. `Kind` label indicates the snapshot kind i.e. full snapshot or incremental/delta snapshot in the context. And succeeded indicates whether the metrics is for successful operation or erroneous operation.
+
+| Name | Description | Type |
+|------|-------------|------|
+| etcdbr_snapshot_duration_seconds | Total latency distribution of saving snapshot to object store. | Histogram |
+| etcdbr_snapshot_gc_total | Total number of garbage collected snapshots. | Counter |
+| etcdbr_snapshot_latest_revision | Revision number of latest snapshot taken. | Gauge |
+| etcdbr_snapshot_latest_timestamp | Timestamp of latest snapshot taken. | Gauge |
```

Comment:
Can we also include some metrics about errors? Something like `etcdbr_snapshot_failure_count` and/or `etcdbr_slapshot_failure_timestamp`?

Also, it would be good to have similar metrics about the delta snapshots. This could be in a different PR also.

Reply:
> Can we also include some metrics about errors? Something like etcdbr_snapshot_failure_count and/or etcdbr_slapshot_failure_timestamp?

On snapshot related metrics, I have applied labels, `succeeded` with which we can scrap the metrics for failure or successful case separately. 

> Also, it would be good to have similar metrics about the delta snapshots. This could be in a different PR also.

To differentiate between delta and full snapshot, i have added label `kind` indicating kind of snapshot.

Since duration is histogram, it adds the count and sum metrics series as well. So, i didn't had to add it explicitly. Though info seems little hidden, I tried to follow the best practices mentioned here.  https://prometheus.io/docs/practices/ which i found therotically optimal for avoiding overhead due to regular metrics scapping.